history/subscriber/kafkaSubscriber.py

```diff
@@ -0,0 +1,236 @@
+import base64
+import logging
+import json
+import time
+from datetime import datetime
+from multiprocessing import Process
+
+import requests
+from kafka import KafkaConsumer
+from kafka.errors import KafkaTimeoutError
+import pymongo
+
+from history import settings
+
+LOGGER = logging.getLogger('history.' + __name__)
+LOGGER.addHandler(logging.StreamHandler())
+LOGGER.setLevel(logging.INFO)
+
+class KafkaEventHandler(object):
+    """
+        Base callback structure for kafka events callbacks
+    """
+    def handle_event(self, message):
+        """
+            Handles a given kafka received message
+            :param message The message that has been received
+        """
+        raise NotImplementedError("Abstract method called")
+
+class TenancyHandler(KafkaEventHandler):
+
+    def __init__(self):
+        pass
+
+    @staticmethod
+    def _spawn_watcher(service, subject, handler):
+        topic = get_topic(service, subject)
+        if topic is None:
+            # TODO how to handle?
```

Comment:
Throw an alarm?

The only way this might happen is when another module failed to initialize the topic, right?

Reply:
Seems like this might also happen when somebody request data from a removed topic.

Reply:
🤔 
Topics are not removed - if a user requests a subject which does not have an associated topic, the "broker" assigns a new one.

This mainly deals with errors when requesting data to the data-broker.

You are correct, though, in pointing out that it may be better to handle errors via exceptions than special return values.

Reply:
now an exception is thrown